### PR TITLE
PRT-1166 cache hash key migration.

### PR DESCRIPTION
**Pull request by ranlavanet**

### CODE REVIEW

File: protocol/rpcprovider/rpcprovider_server.go [hunk 1]
```diff
@@ -766,7 +781,21 @@ func (rpcps *RPCProviderServer) TryRelay(ctx context.Context, request *pairingty
...
+					return
+				}
+				err = cache.SetEntry(new_ctx, &pairingtypes.RelayCacheSet{
+					RequestHash:      hash,
+					RequestedBlock:   requestedBlock,
+					BlockHash:        requestedBlockHash,
+					ChainId:          rpcps.rpcProviderEndpoint.ChainID,
+					Response:         copyReply,
+					Finalized:        finalized,
+					OptionalMetadata: ignoredMetadata,
```
Comment (omerlavanet) [inline protocol/rpcprovider/rpcprovider_server.go:797]: you forgot average block time and seen block

File: protocol/rpcprovider/rpcprovider_server.go [hunk 2]
```diff
@@ -722,13 +722,28 @@ func (rpcps *RPCProviderServer) TryRelay(ctx context.Context, request *pairingty
...
+		if hashErr != nil {
+			utils.LavaFormatError("TryRelay Failed computing hash for cache request", hashErr)
+		} else {
+			cacheCtx, cancel := context.WithTimeout(ctx, common.CacheTimeout)
+			cacheReply, err = cache.GetEntry(cacheCtx, &pairingtypes.RelayCacheGet{
+				RequestHash:    hashKey,
+				RequestedBlock: request.RelayData.RequestBlock,
+				ChainId:        rpcps.rpcProviderEndpoint.ChainID,
+				BlockHash:      requestedBlockHash,
+				Finalized:      finalized,
```
Comment (omerlavanet) [inline protocol/rpcprovider/rpcprovider_server.go:736]: seen block missing here

**Review by omerlavanet — CHANGES_REQUESTED** [review]

**Review by omerlavanet — CHANGES_REQUESTED** [review]
let's add the following test too:
set up a provider, cache using chainFetcher, see that we hit the cache by requesting from a user (with seen block)

File: protocol/rpcprovider/rpcprovider_server.go (hunk 2 above)
Comment (ranlavanet) [reply protocol/rpcprovider/rpcprovider_server.go:736]: agreed that its ok.

**Review by omerlavanet — APPROVED** [review]

**Pull request merged**